spec/unit/lib/wifi_user/use_cases/sponsor_users_spec.rb

```diff
@@ -148,6 +156,7 @@ def a_plural_sponsor_confirmation_request
     let(:sponsor) { 'adrian <adrian@fake.uk>' }
     let(:sponsees) { ['adrian@notgov.uk'] }
     let(:do_not_reply_id) { staging_do_not_reply_id }
+    let(:whitelist_checker) { double(execute: { success: false }) }
```

Comment:
The test structure feels a bit weird to me with the changes to whitelist_checker.

The test reads half like it's an acceptance spec:
- Here's an email that wouldn't work in the real world
- Try out the process
- See behaviour

And half like it's a unit spec with mocking collaborators
- Here's some irrelevant input
- Here's a collaborator that always returns false
- Try out the process
- See behaviour

Maybe some of this setup doesn't need to be here? Or it can be expressed more tersely?

Finding this hard to explain in text, though.

Reply:
Agreed, once again I don't think this was introduced by this new whitelist feature.

Code needs to be looked at again as a whole, and some classes teased apart.